go.mod

```diff
@@ -10,7 +10,7 @@ require (
 	github.com/cockroachdb/pebble v1.1.2
 	github.com/cosmos/go-bip39 v1.0.0
 	github.com/dgraph-io/badger/v4 v4.4.0
-	github.com/dgraph-io/ristretto v1.0.0
```

Comment:
we are downgrading the ristretto version, should we not upgrade to [v2.0.0](https://github.com/dgraph-io/ristretto/releases/tag/v2.0.0)?

Reply:
The hint in the warning is misleading in this case, since `go get github.com/dgraph-io/ristretto@latest` will only fetch versions below v2. Go modules with major versions above 1 [need to have a different import path](https://go.dev/ref/mod#major-version-suffixes) (in this case `github.com/dgraph-io/ristretto/v2`).

Turns out we already have `github.com/dgraph-io/ristretto/v2` as [an indirect dependency in `go.mod`](https://github.com/ChainSafe/gossamer/blob/4e6f8c4b22c2afa7d79d3f2e7764c78fecc60add/go.mod#L73). It came in with the upgrade of `github.com/dgraph-io/badger/v4` [a few days ago](https://github.com/ChainSafe/gossamer/pull/4331/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R73).

So what we should do in this PR is add `v2` to the import paths in all files that use Ristretto and run `go mod tidy` to remove the reference to `github.com/dgraph-io/ristretto` and change the reference to `github.com/dgraph-io/ristretto/v2` to a direct dependency.

FWIW, when I just did that on current `development`, I did not need the code updates in this PR.